client/pages/Dashboard/index.js

```diff
@@ -118,14 +119,56 @@ class Dashboard extends Component {
   }
 }
 
+Dashboard.defaultProps = {
+  isAuthenticated: false,
+  cbOpenLoginModal: undefined,
+  curUser: undefined,
+  events: undefined,
+  cbDeleteEvent: undefined,
+  cbInviteEmail: undefined,
+};
+
 Dashboard.propTypes = {
   isAuthenticated: PropTypes.bool,
   cbOpenLoginModal: PropTypes.func,
-  curUser: PropTypes.object,
-  events: PropTypes.array,
+
+  // Current user
+  curUser: PropTypes.shape({
+    _id: PropTypes.string,      // Unique user id
+    name: PropTypes.string,     // User name
+    avatar: PropTypes.string,   // URL to image representing user(?)
+  }),
+
+  events: PropTypes.arrayOf(
+    PropTypes.shape({
+      _id: PropTypes.string,
+      name: PropTypes.string,
+      owner: PropTypes.string,
+      active: PropTypes.bool,
+      selectedTimeRange: PropTypes.array,
+      dates: PropTypes.arrayOf(PropTypes.shape({
+        fromDate: PropTypes.string,
+        toDate: PropTypes.string,
+        _id: PropTypes.string,
+      })),
+      participants: PropTypes.arrayOf(PropTypes.shape({
+        userId: PropTypes.shape({
+          id: PropTypes.string,
+          avatar: PropTypes.string,
+          name: PropTypes.string,
+          emails: PropTypes.arrayOf(PropTypes.string),
+        }),
+        _id: PropTypes.string,
+        status: PropTypes.oneOf([0, 1, 2, 3]),
+        emailUpdate: PropTypes.bool,
+        ownerNotified: PropTypes.bool,
+        availability: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)),
+      })),
+    }),
+  ),
+
   cbDeleteEvent: PropTypes.func,
-  cbDeleteGuest: PropTypes.func,
-  cbInviteEmail: React.PropTypes.func,
+  cbInviteEmail: PropTypes.func,
```

Comment:
if is not required set as default like:  editAvail: () => { console.log('ediAvail func not passed in!'); },